test/hotspot/jtreg/runtime/8176717/TestInheritFD.java

```diff
@@ -290,8 +305,11 @@ public static void main(String[] args) throws InterruptedException {
                 if (false) {  // Enable to simulate a timeout in the third VM.
                     Thread.sleep(300 * 1000);
                 }
+            } catch (CompletionException e) {
+                System.out.println("(Third VM) Timed out waiting for second VM: " + e.toString());
```

Comment:
You don't know it timed out. The cause of the `CompletionException` may be something else. You should check the cause and then do your timeout message, else report an unexpected completion exception.

Reply:
Thank you, I check if it's a `TimeoutException` that's the cause, and if so print that a timeout occurred, otherwise print the 'regular' exception message.